src/AbstractFactory/ReflectionBasedAbstractFactory.php

```diff
@@ -10,6 +10,7 @@
 
 use Laminas\ServiceManager\Exception\ServiceNotFoundException;
 use Laminas\ServiceManager\Factory\AbstractFactoryInterface;
+use Laminas\Stdlib\DispatchableInterface;
```

Comment:
This is a mistake for
```
    /**
     * {@inheritDoc}
     *
     * @return DispatchableInterface
     */
    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
```

Not returned DispatchableInterface, just returned object  - requestedName instance